MoTrPAC/motrpac-frontend
# Resolves #92 ct circular redirect

# **Key Changes**
1. Users who are either not authenticated or don't have access to the data are now routed to the error page and not the homepage
2. Users routed to the error page should not be able to see a login button (this is the issue with the circular redirect). Instead, a "Return to home" button has been added which will return the user to the homepage where they can attempt login again

## **To Test**
1. Create/use a user who lacks user metadata and attempt to login (either through Google Auth or username/password). You should be routed to the error page without a login button.
2. Create/use a user who does not have access to the data (either through Google Auth or username/password). You should be routed to the error page without a login button.
3. Create/use a user who does have access to the data. You should be routed to the dashboard.

## **Known Issues**
1. There is a brief period where the dashboard pops up when being routed to the error page.
2. If you navigate back to the homepage directly (by URL) without using the "Return to home" button, the login button will not be available in either the navigation bar or the footer. This is because Auth0 considers the user as still "authenticated" but without access, and therefore the login button doesn't appear for them.

<img width="1640" alt="Screen Shot 2020-12-03 at 11 51 28 AM" src="https://user-images.githubusercontent.com/29579769/101080720-f8f57a80-355d-11eb-95a9-ec454bdc155d.png">
